v2.0.x release preparation

Thoemi09 opened the pull request:
- update/format doc strings and doc pages
- remove compiler warnings and other minor code simplifications
- move examples codes present in the docs to an examples folder and make sure that they compile
- port to c2py + clair
  - remove all cpp2py references
  - clean up h5py_io library

Thoemi09 committed:
Update Doxyfile to v1.14.0

Thoemi09 committed:
Format doc strings

Thoemi09 committed:
Remove compiler warnings and minor simplifications

Thoemi09 committed:
Increase version to 2.0.0 and update docs

Thoemi09 committed:
Remove ChangeLog.md file from c++/h5

Thoemi09 committed:
Use requires instead of H5_REQUIRES macro

Thoemi09 committed:
Add example codes to doc folder and make sure they compile

Thoemi09 committed:
Remove unused pybind11 bindings

Thoemi09 committed:
Move static functions to anonymous namespace in h5py_io.cpp

Thoemi09 committed:
Simplify mapping between h5 and numpy data types in h5py_io.cpp

Thoemi09 committed:
Remove support for numpy versions < 1.17

Thoemi09 committed:
General clean up in h5py_io.cpp

- use consistent variable names
- basic modernization, e.g. NULL -> nullptr, remove C-style casts, etc.

Thoemi09 committed:
Move Storable concept into its own header and add tests

Thoemi09 committed:
Use "Tuple" as a format string for std::pair and std::tuple

Thoemi09 committed:
Add hdf5_format_impl specialization for char_buf

Thoemi09 committed:
Add additional group methods used to generate c2py/clair bindings

Thoemi09 committed:
Add serialization.hpp to h5.hpp

Thoemi09 committed:
Port h5 to c2py + clair

- remove all cpp2py references
- h5py_io
  - return pyref instead of PyObject * to avoid clair issue
  - include specific h5 headers to avoid problems with _h5py.wrap.hxx
  - do not call import_array() --> called only once in _h5py.wrap.cxx as described in numpy docs
  - add doxygen docs
- change python layer to account
- update tests
- - remove generated header from include dir, install it directly into include/h5 instead

Thoemi09 committed:
Update documentation

- fix typos
- remove explicit version requirements for dependencies

Thoemi09 committed:
Add Python interface utilities to doxygen

Thoemi09 committed:
Update documentation of python layer

Thoemi09 committed:
Add generated python bindings

Thoemi09 committed:
Remove stale ChangeLog.md

Thoemi09 committed:
Update changelog for v2.0.0

Wentzell reviewed (approved):
Thank you @Thoemi09 for this big update

Wentzell committed:
Update CI workflow for c2py/clair port

Build and install clair from source so the c2py bindings can be regenerated.
Switch to versioned LLVM-20 toolchain on Ubuntu and drop cpp2py-era deps
(mako, MPI, fftw, etc.) that h5 does not use.

Assisted-by: Claude <noreply@anthropic.com>
